src/Foundation/NSUrlSessionHandler.cs

```diff
@@ -335,6 +337,16 @@ protected override void Dispose (bool disposing)
 			}
 		}
 
+		public CookieContainer CookieContainer {
```

Comment:
[AndroidClientHandler.cs](https://github.com/xamarin/xamarin-android/blob/master/src/Mono.Android/Xamarin.Android.Net/AndroidClientHandler.cs#L853) also has a `UseCookies` property inherited from [HttpClientHandler](https://docs.microsoft.com/en-us/dotnet/api/system.net.http.httpclienthandler.usecookies?view=netframework-4.8). Interestingly enough, `NSUrlSessionHandler` inherits from `HttpMessageHandler` instead :-?

I'd say the property is useful since it allows you to opt-out from this feature.

Reply:
I can implement such a property BUT we should do it in a diff PR. The issue is that we also have to deal with the fact that iOS has its own 'CookieContainer' so if we set the property, we have to make sure that we do not use the native CookieContainer or the managed one. I'd like to keep the PR small since this code is complicated. Added an issue that I'll fix once we land this: https://github.com/xamarin/xamarin-macios/issues/7659